pkg/agent/sysadvisor/plugin/qosaware/resource/cpu/advisor.go

```diff
@@ -428,14 +437,48 @@ func (cra *cpuResourceAdvisor) assembleProvision() (InternalCalculationResult, e
 
 	// Fill in reclaimed pool size of non binding numas
 	reservePoolSizeOfNonBindingNumas := int(math.Ceil(float64(reservePoolSize*cra.nonBindingNumas.Size()) / float64(cra.metaServer.NumNUMANodes)))
-	reclaimPoolSizeOfNonBindingNumas := cra.nonBindingNumas.Size()*cra.metaServer.CPUsPerNuma() - nonNumaBindingRequirement - reservePoolSizeOfNonBindingNumas
+
+	reclaimPoolSizeOfNonBindingNumas := general.Max(cra.nonBindingNumas.Size()*cra.metaServer.CPUsPerNuma()-nonNumaBindingRequirement-reservePoolSizeOfNonBindingNumas,
```

Comment:
can we separate this function to several different functions? for instance, one to assemble dedicated pools, one for shared pools, one for reclaimed, and so on; just a proposal, you may raise up with a better separating way